generative_models/ContextEncoder_PyTorch/main.py

```diff
@@ -0,0 +1,171 @@
+        ('base_d_lr', args.base_d_lr),
+        ('beta_1', args.beta_1),
+        ('beta_2', args.beta_2),
+    ])
+
+    run_config = OrderedDict([
+        ('device', args.device),
+        ('num_workers', args.num_workers),
+        ('label_smoothing', args.label_smoothing),
+        ('test_batch_size',args.test_batch_size),
+        ('writer_directory',args.writer_directory),
+        ('save_frequency',args.save_frequency),
+    ])
+
+    data_config = OrderedDict([
+         ('dataset_name', args.dataset_name),
+         ('mask_size', args.mask_size),
+        ])
+
+    model_config = OrderedDict([
+        ('in_channels',args.in_channels),
+        ])
+    
+    config = OrderedDict([
+        ('optim_config', optim_config),
+        ('data_config', data_config),
+        ('run_config', run_config),
+        ('model_config', model_config),
+    ])
+
+    return config
+
+config = parse_args()
+
+run_config = config['run_config']
+optim_config = config['optim_config']
+device = run_config['device']
+model_config = config['model_config']
+dataset_name = config['data_config']['dataset_name']
+nc = model_config['in_channels']
+mask_size = config['data_config']['mask_size']
+
+#get dataloaders
+dataloader,test_dataloader = get_loader(optim_config['train_batch_size'],run_config['test_batch_size'],mask_size,dataset_name,run_config['num_workers'])
+#loss functions
+adversarial_loss = torch.nn.BCEWithLogitsLoss()
+pixelwise_loss = torch.nn.MSELoss()
+
+#models
+generator = Generator(nc).to(device)
+discriminator = Discriminator(nc).to(device)
+generator.apply(weights_init_normal)
+discriminator.apply(weights_init_normal)
+
+#optimizers 
+optimizer_G = torch.optim.Adam(generator.parameters(), lr=optim_config['base_g_lr'], betas=(optim_config['beta_1'],optim_config['beta_2']))
+optimizer_D = torch.optim.Adam(discriminator.parameters(), lr=optim_config['base_d_lr'], betas=(optim_config['beta_1'],optim_config['beta_2']))
+
+#run configurations
+epochs = optim_config['epochs']
+writer = SummaryWriter(run_config['writer_directory'])
+
+#training loop
+for epoch in range(epochs):
+    
+    #load tqdm object
+    t = tqdm(dataloader, desc='epoch:{} g_loss:{:.4f} d_loss:{:.4f}'.format(epoch, 0.0, 0.0), leave=True)
+    
+    for i, (imgs, masked_imgs, masked_parts) in enumerate(t):
+
+        # Adversarial ground truths
+        valid = real_data_target(imgs.size(0),'cuda')
+        fake = fake_data_target(imgs.size(0),'cuda')
+        imgs = imgs.to(device)
+        masked_imgs = masked_imgs.to(device) 
+        masked_parts = masked_parts.to(device)
+
+        #current step used for logging
+        current_step = epoch*len(dataloader) + i
+
+        #train generator
+        optimizer_G.zero_grad()
+
+        gen_parts = generator(masked_imgs)
+
+        #calculate loss
+        g_adv = adversarial_loss(discriminator(gen_parts), valid)
+        g_pixel = pixelwise_loss(gen_parts, masked_parts)      
+        g_loss = 0.001 * g_adv + 0.999 * g_pixel
```

Comment:
```suggestion
        g_loss = lambda_adv * g_adv + (1 - lambda_adv) * g_pixel
```
and add lambda_adv to parser arg